test/test_curl.cpp

```diff
@@ -9,103 +9,131 @@
 #include <chrono>
 #include <exception>
 #include <system_error>
+#include <unordered_set>
 
 #include "mocks/loggers.h"
 #include "test.h"
 
 using namespace datadog::tracing;
 
-TEST_CASE("parse response headers and body") {
-  class MockCurlLibrary : public CurlLibrary {
-    void *user_data_on_header_ = nullptr;
-    HeaderCallback on_header_ = nullptr;
-    void *user_data_on_write_ = nullptr;
-    WriteCallback on_write_ = nullptr;
-    CURL *handle_ = nullptr;
-    CURLMsg message_;
-
-    CURLcode easy_getinfo_response_code(CURL *, long *code) override {
-      *code = 200;
-      return CURLE_OK;
-    }
-    CURLcode easy_setopt_headerdata(CURL *, void *data) override {
-      user_data_on_header_ = data;
-      return CURLE_OK;
-    }
-    CURLcode easy_setopt_headerfunction(CURL *,
-                                        HeaderCallback on_header) override {
-      on_header_ = on_header;
-      return CURLE_OK;
-    }
-    CURLcode easy_setopt_writedata(CURL *, void *data) override {
-      user_data_on_write_ = data;
-      return CURLE_OK;
-    }
+class SingleRequestMockCurlLibrary : public CurlLibrary {
```

Comment:
This code is moved and renamed from the `class MockCurlLibrary` that used to be in the `TEST_CASE("parse response headers and body")`. It's used there and in the new test case.

I also added data members `created_handles_`, `destroyed_handles_`, `message_result_`, and `delay_message_`.